packages/side-nav/src/SideNavGroupCollapsed/SideNavGroupCollapsed.tsx

```diff
@@ -46,21 +46,27 @@ export function SideNavGroupCollapsed({
   const { width, theme, darkMode } = useSideNavContext();
   const { usingKeyboard } = useUsingKeyboardContext();
 
+  const [ulHeight, setUlHeight] = useState(0);
+
   const menuId = useIdAllocator({ prefix: 'menu' });
 
   const nodeRef = React.useRef(null);
-  const ulRef = React.useRef<HTMLUListElement>(null);
+
+  const ulRef = useCallback((node: HTMLUListElement) => {
```

Comment:
how interesting! i was thinking of something else, like this:
```
const [ulRef, setUlRef] = useState(0);
```
```
ref={setUlRef} // when the ref is available the state will update
```
```
useEffect(() => {
    const ulHeight = ulRef?.getBoundingClientRect().height ?? 0;
  }, [open, ulRef, darkMode]);
```
But if your approach works the same way, it should be good. 
When the ref changes, does that trigger `useCallback`?